sourcegraph/sourcegraph
# bext: add firstSourcegraphUrl to logs

Closes #32249

Adds `firstSourcegraphUrl` to logs when default Sourcegraph URL is used.

## Notes
- tested in Chrome
- telemetry is disabled for Firefox, so logs are not sent at all, see:
  - [client/browser/src/shared/code-hosts/shared/codeHost.tsx](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/browser/src/shared/code-hosts/shared/codeHost.tsx?L1558-1568)
  - [client/browser/src/shared/util/optionFlags.ts](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/browser/src/shared/util/optionFlags.ts?L91-123)
 
## Test plan

1. Run  `sg run bext` for _dev_ version or `yarn --cwd client/browser build` for _prod_ one.
2. Set Sourcegrapg URL value to _sourcegraph.com_.
3. Open a file on a code host (for example, [this one](https://github.com/sourcegraph/sourcegraph/blob/main/client/browser/src/browser-extension/options-menu/OptionsPage.tsx)).
4. Open browser extension background page.
5. Perform an action being logged (for example, hover on a token to see a code intel popup).
6. Inspect `logEvent` requests on the network tab of browser extension background page.
7. Ensure that `firstSourceUrl` value **is present** in the request payload.
8. Change Sourcegrapg URL value (for example, to _k8s.sgdev.org_).
9. Repeat steps 5 and 6.
10. Ensure that `firstSourceUrl` value **is not present** in the request payload.
